### PR TITLE
[EasyCore] Added listener for worker restart on EM close

### PR DESCRIPTION
**Pull request by skrut**
- added listener for worker restarting

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #PTS-962

### CODE REVIEW

File: packages/EasyCore/src/Bridge/Laravel/Providers/EasyCoreServiceProvider.php [hunk 1]
```diff
@@ -44,4 +47,18 @@ private function logQueueWorkerStopping(): void
 
         $this->app->get('events')->listen(WorkerStopping::class, QueueWorkerStoppingListener::class);
     }
+
+    /**
+     * Restarts Queue.
+     *
+     * @return void
+     */
+    private function restartQueue(): void
```
Comment (roman-loyaltycorp) [inline packages/EasyCore/src/Bridge/Laravel/Providers/EasyCoreServiceProvider.php:56]: Please rename to `restartQueueOnEmClose` and remove the method description (to make the code similar to the existing one).

File: packages/EasyCore/tests/Bridge/Laravel/Listeners/DoctrineRestartQueueOnEmCloseListenerTest.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ * @internal
+ */
+final class DoctrineRestartQueueOnEmCloseListenerTest extends AbstractTestCase
+{
+    /**
+     * Test `handle` succeeds.
+     *
+     * @return void
+     */
+    public function testHandleSucceeds(): void
```
Comment (roman-loyaltycorp) [inline packages/EasyCore/tests/Bridge/Laravel/Listeners/DoctrineRestartQueueOnEmCloseListenerTest.php:29]: How about testing a case when EM is not closed?

File: packages/EasyCore/src/Bridge/Laravel/Providers/EasyCoreServiceProvider.php (hunk 1 above)
Comment (skrut) [reply packages/EasyCore/src/Bridge/Laravel/Providers/EasyCoreServiceProvider.php]: Done.

File: packages/EasyCore/tests/Bridge/Laravel/Listeners/DoctrineRestartQueueOnEmCloseListenerTest.php (hunk 2 above)
Comment (skrut) [reply packages/EasyCore/tests/Bridge/Laravel/Listeners/DoctrineRestartQueueOnEmCloseListenerTest.php:30]: Done.

**Pull request merged**